### PR TITLE
ARTEMIS-1187 Incompatible version when recreating a session with olde…

### PR DESCRIPTION
**Pull request by TomasHofman**
…r server

https://issues.apache.org/jira/browse/ARTEMIS-1187
https://issues.jboss.org/browse/JBEAP-9522

### CODE REVIEW

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java [hunk 1]
```diff
@@ -686,7 +686,7 @@ protected CreateSessionMessage newCreateSession(String username,
                                                    boolean autoCommitSends,
                                                    boolean autoCommitAcks,
                                                    boolean preAcknowledge) {
-      return new CreateSessionMessage(name, sessionChannel.getID(), VersionLoader.getVersion().getIncrementingVersion(), username, password, minLargeMessageSize, xa, autoCommitSends, autoCommitAcks, preAcknowledge, confirmationWindow, null);
+      return new CreateSessionMessage(name, sessionChannel.getID(), serverVersion, username, password, minLargeMessageSize, xa, autoCommitSends, autoCommitAcks, preAcknowledge, confirmationWindow, null);
```
Comment (clebertsuconic) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java]: I'm not sure about this... the purpose is the client to tell the server which version is being used... it looks wrong to me.

Comment (TomasHofman) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java]: The ```newCreateSession()``` is only called in ```recreateSession()``` method, so my reasoning was that the message will be sent to the same server that the client was previously connected to (is that correct?), so the serverVersion that was previously established can be reused.

The alternative could be to start a cycle of CreateSessionMessages with the highest know version and decrease the version if ```ActiveMQIncompatibleClientServerException``` is returned? I think this is how establishing new connection works.

Comment (clebertsuconic) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java]: Ohhh.. I see.. you're right!


merging it.. sorry for the hassle.. just wanted to make sure what was going on.

Comment (TomasHofman) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java]: Thanks, I'm glad for feedback, I might have also be wrong...

**Pull request closed without merging**